x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/ClaimConnectorSyncJobAction.java

```diff
@@ -0,0 +1,160 @@
+        private final Object syncCursor;
+
+        public String getConnectorSyncJobId() {
+            return connectorSyncJobId;
+        }
+
+        public String getWorkerHostname() {
+            return workerHostname;
+        }
+
+        public Object getSyncCursor() {
+            return syncCursor;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.connectorSyncJobId = in.readString();
+            this.workerHostname = in.readString();
+            this.syncCursor = in.readGenericValue();
+        }
+
+        public Request(String connectorSyncJobId, String workerHostname, Object syncCursor) {
+            this.connectorSyncJobId = connectorSyncJobId;
+            this.workerHostname = workerHostname;
+            this.syncCursor = syncCursor;
+        }
+
+        private static final ConstructingObjectParser<Request, String> PARSER = new ConstructingObjectParser<>(
+            "claim_connector_sync_job",
+            false,
+            (args, connectorSyncJobId) -> {
+                String workerHostname = (String) args[0];
+                Object syncCursor = args[1];
+
+                boolean hasSyncCursor = syncCursor != null;
+
+                return new Request(connectorSyncJobId, workerHostname, hasSyncCursor ? syncCursor : null);
+            }
+        );
+
+        static {
+            PARSER.declareString(constructorArg(), ConnectorSyncJob.WORKER_HOSTNAME_FIELD);
+            PARSER.declareObject(optionalConstructorArg(), (parser, context) -> parser.map(), Connector.SYNC_CURSOR_FIELD);
+        }
+
+        public static Request fromXContent(XContentParser parser, String connectorSyncJobId) throws IOException {
+            return PARSER.parse(parser, connectorSyncJobId);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            {
+                builder.field(ConnectorSyncJob.WORKER_HOSTNAME_FIELD.getPreferredName(), workerHostname);
+                builder.field(Connector.SYNC_CURSOR_FIELD.getPreferredName(), syncCursor);
+            }
+            builder.endObject();
+            return builder;
+        }
+
+        public static Request fromXContentBytes(String connectorSyncJobId, BytesReference source, XContentType xContentType) {
+            try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, source, xContentType)) {
+                return fromXContent(parser, connectorSyncJobId);
+            } catch (IOException e) {
+                throw new ElasticsearchParseException("Failed to parse request" + source.utf8ToString());
+            }
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(connectorSyncJobId)) {
```

Comment:
https://github.com/elastic/connectors/blob/fd0f9431682cd79ba51e474879904b1347d6beab/connectors/protocol/connectors.py#L332 

I think we should accept empty string, as [gethostname()](https://docs.python.org/3/library/socket.html#socket.gethostname) just returns string and doesn't guarantee it's not empty